File: math/programmers_secret_map.java
/*
테스트 1 〉	통과 (2.18ms, 74.1MB)
테스트 2 〉	통과 (2.59ms, 77.6MB)
테스트 3 〉	통과 (1.92ms, 78.3MB)
테스트 4 〉	통과 (2.55ms, 75.7MB)
테스트 5 〉	통과 (2.16ms, 77.1MB)
테스트 6 〉	통과 (2.41ms, 74.1MB)
테스트 7 〉	통과 (2.07ms, 75.5MB)
테스트 8 〉	통과 (2.00ms, 78.4MB)
*/

class Solution {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer = new String[n];
        String[] str1 = new String[n];
        String[] str2 = new String[n];
        
        for (int i = 0; i < n; i++) {
            int num = arr1[i];
            String s = "";
            
            while (num > 0) {
                s = Integer.toString(num%2) + s;
                num /= 2;
            }
            
            while (s.length() < n) s = "0" + s;
            str1[i] = s;
        }
        
        for (int i = 0; i < n; i++) {
            int num = arr2[i];
            String s = "";
            
            while (num > 0) {
                s = Integer.toString(num%2) + s;
                num /= 2;
            }
            
            while (s.length() < n) s = "0" + s;
            str2[i] = s;
        }
        
        for (int i = 0; i < n; i++) {
            String s = "";
            
            for (int j = 0; j < n; j++) {
                s += ((str1[i].charAt(j) == '1' || str2[i].charAt(j) == '1') ? "#" : " ");
            }
            answer[i] = s;
        }
        
        return answer;
    }
}

File: math/programmers_secret_map.cpp
/*
테스트 1 〉	통과 (0.02ms, 3.61MB)
테스트 2 〉	통과 (0.02ms, 4.25MB)
테스트 3 〉	통과 (0.01ms, 4.26MB)
테스트 4 〉	통과 (0.02ms, 4.23MB)
테스트 5 〉	통과 (0.01ms, 4.25MB)
테스트 6 〉	통과 (0.03ms, 4.25MB)
테스트 7 〉	통과 (0.02ms, 3.69MB)
테스트 8 〉	통과 (0.01ms, 4.2MB)
*/

#include <string>
#include <vector>

using namespace std;

vector<string> solution(int n, vector<int> arr1, vector<int> arr2) {
	vector<string> answer;

	for (int i = 0; i < n; i++) {
        string ans = "";
        arr1[i] = arr1[i]|arr2[i];
        
        for (int j = 0; j < n; j++) {
            if (arr1[i] % 2 == 0) ans = ' ' + ans;
            else ans = '#' + ans;
            
            arr1[i] = arr1[i] >> 1;
        }
        
		answer.push_back(ans);
	}

	return answer;
}
